DQMOffline/Muon/src/TriggerMatchMonitor.cc

```diff
@@ -0,0 +1,216 @@
+#include "DQMServices/Core/interface/MonitorElement.h"
+#include "DataFormats/Common/interface/TriggerResults.h"
+#include "FWCore/Common/interface/TriggerNames.h"
+#include "TLorentzVector.h"
+
+#include <string>
+#include <TMath.h>
+using namespace std;
+using namespace edm;
+
+//#define DEBUG
+
+TriggerMatchMonitor::TriggerMatchMonitor(const edm::ParameterSet& pSet) {
+  LogTrace(metname)<<"[TriggerMatchMonitor] Parameters initialization";
+  
+  parameters = pSet;
+
+  // the services
+  theService = new MuonServiceProxy(parameters.getParameter<ParameterSet>("ServiceParameters"));
+  theDbe = edm::Service<DQMStore>().operator->();
+  
+  beamSpotToken_ = consumes<reco::BeamSpot >(parameters.getUntrackedParameter<edm::InputTag>("offlineBeamSpot")),
+  primaryVerticesToken_ = consumes<std::vector<reco::Vertex> >(parameters.getUntrackedParameter<edm::InputTag>("offlinePrimaryVertices")),
+  theMuonCollectionLabel_ = consumes<edm::View<reco::Muon> >  (parameters.getParameter<edm::InputTag>("MuonCollection"));
+  thePATMuonCollectionLabel_ = consumes<edm::View<pat::Muon> >  (parameters.getParameter<edm::InputTag>("patMuonCollection"));
+  theVertexLabel_          = consumes<reco::VertexCollection>(parameters.getParameter<edm::InputTag>("VertexLabel"));
+  theBeamSpotLabel_        = mayConsume<reco::BeamSpot>      (parameters.getParameter<edm::InputTag>("BeamSpotLabel"));
+  triggerResultsToken_ = consumes<edm::TriggerResults>(parameters.getUntrackedParameter<edm::InputTag>("triggerResults"));
+  triggerObjects_ = consumes<std::vector<pat::TriggerObjectStandAlone>>(parameters.getParameter<edm::InputTag>("triggerObjects"));
+
+  triggerPathName1_ = parameters.getParameter<string>("triggerPathName1");
+  triggerHistName1_ = parameters.getParameter<string>("triggerHistName1");
+  triggerPtThresholdPath1_ = parameters.getParameter<double>("triggerPtThresholdPath1");
+  triggerPathName2_ = parameters.getParameter<string>("triggerPathName2");
+  triggerHistName2_ = parameters.getParameter<string>("triggerHistName2");
+  triggerPtThresholdPath2_ = parameters.getParameter<double>("triggerPtThresholdPath2");
+  theFolder = parameters.getParameter<string>("folder");
+}
+TriggerMatchMonitor::~TriggerMatchMonitor() { 
+  delete theService;
+}
+
+void TriggerMatchMonitor::bookHistograms(DQMStore::IBooker & ibooker,
+					  edm::Run const & /*iRun*/,
+					  edm::EventSetup const& /*iSetup*/){
+  ibooker.cd();
+  ibooker.setCurrentFolder(theFolder);
+
+  // monitoring of trigger match parameter
+   
+  matchHists.push_back(ibooker.book1D("DelR_HLT_"+triggerHistName1_+"_v1", "DeltaR_(offline,HLT)_triggerPass("+triggerHistName1_+")", 500, 0.0, 0.5));
+  matchHists.push_back(ibooker.book1D("DelR_HLT_"+triggerHistName1_+"_v2", "DeltaR_(offline,HLT)_triggerPass("+triggerHistName1_+")", 1000, 0.0, 1.0));
+  matchHists.push_back(ibooker.book1D("DelR_HLT_"+triggerHistName1_+"_v3", "DeltaR_(offline,HLT)_triggerPass("+triggerHistName1_+")", 5000, 0.0, 5.0));
```

Comment:
@fabiocos Yes, for most cases if the matching is correct, one would expect small values of delta R. To see this distribution the range has been set small. But in case the matching breaks it could have large values of delta R. In fact this we want to monitor if the matching works fine or not. Making only one large range histogram makes it difficult to see the histogram shape in low range. So added different versions of same histogram.

Reply:
@bmahakud sorry, but the bin width is the same in all the variants, so what is the real advantage? I might understand better if the larger ranges have a coarser binning. The memory cost of this PR is not huge, but I wonder whether it is really justified.

Reply:
@fabiocos @folguera
I thought one cannot zoom-in/out the histograms in the GUI. So in a range of 0 to 5, if some body wants to see the shape within 0 to 0.3 it is not visible as you could see in one of the sample histograms here . 
http://bmahakud.web.cern.ch/bmahakud/TriggerMatchMonitorHistos/DeltaR_offlineHLT_passMu50_v2.pdf
If zoom option is available in the GUI, then I think this addition of histograms are unnecessary. I am sorry if there is an option available and I do not know. Anyway, if you are not happy , I can keep only one version with range from 0 to 5. 

@folguera Please comment what do you think?

Reply:
@bmahakud I do not want to make a big deal out of this, but: do you really need to have the same binning in the 0-5 range as in the 0-0.5 range? If you integrate more on the larger range you can easily see them without zoom, otherwise the visibility problems you have in the low range will happen also in the high tails...

Reply:
@fabiocos Thanks for suggestion. Changed bindings have been committed .

Reply:
@fabiocos are the changes proposed in the most recent commit satisfactory?